### PR TITLE
DEVOPS-3041 authorization middleware

### PR DESCRIPTION
**Pull request by dskatz**
This adds an Authorization middleware.  Note, this still needs to be enabled in the server itself.  An example exists commented out and will be included in a followup PR.

I have also added the same logging middleware used in the k8s-auth-portal and server functional options.   (There will be followup PRs to the kanopy-app-go template repo).

As per spec, validation of the JWT is not performed at this service level. It is relying on an upstream proxy to provide token validation.

The authorization middleware will simply check for an auth token, aggregate the claims (groups and scopes) and pass that information to the request's Context.

It is up to each route to perform an authorization check using the Config controller.

Note, the `go.mod` file includes aws SDK entries to resolve conflicts from the `main` branch.  Running `go mod tidy` locally will try to remove them.  Don't do that.

### CODE REVIEW

File: internal/server/middleware/authorization/authorization.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+type Authorizer interface {
+	Authz(next http.Handler) http.Handler
+}
+
+type Middleware struct {
+	entitlementManager Entitler
+	authCookieName     string
+	authHeaderEnabled  bool
+}
```
Comment (colinhoglund) [inline internal/server/middleware/authorization/authorization.go:31]: It's unclear why this is designed like this. Middleware is typically either a simple function that decorates an `http.Handler` ([as per our style guide](https://github.com/10gen/infrastructure-handbook/blob/master/docs/code/style/docs/go.md#server-example)), or possibly implements [Handler](https://pkg.go.dev/net/http#Handler) itself by defining a `ServeHTTP([ResponseWriter](https://pkg.go.dev/net/http#ResponseWriter), *[Request](https://pkg.go.dev/net/http#Request))` method.

Also, `Authorizer` is being defined and implemented in the same package, which is typically frowned upon, and seems like maybe instead `Middleware` just needs to implement `http.Handler`?

Can you please add some more detailed explanations as to what these interfaces are for and why things are designed this why?

Comment (dskatz) [reply internal/server/middleware/authorization/authorization.go]: The authorizer interface was essentially dead code.    Removed that.

The main reason this was removed from the server code is to encapsulate the complex authorization logic and offer clear interfaces for other internal packages.

That being said, I did push up new commits that only expose a middleware handler function.  I think that is better given that all the members of the `middleware` struct are internal.

**Review by yuzhouliu9 — APPROVED** [review]
LGTM

**Review by whwalter — APPROVED** [review]

File: internal/server/server.go [hunk 2]
```diff
@@ -41,3 +55,39 @@ func (s *Server) handleHealthz() http.HandlerFunc {
 		fmt.Fprint(w, string(bytes))
 	}
 }
+
+func logRequestHandler(next http.Handler) http.Handler {
```
Comment (colinhoglund) [inline internal/server/server.go:59]: suggestion: standard logging middleware seems like something we can add to our app template, or perhaps even a separate http middleware library.

Comment (dskatz) [reply internal/server/server.go:59]: I plan to have follow up PRs for the app template.

**Pull request merged**